indico/modules/events/registration/fields/simple.py

```diff
@@ -371,3 +371,50 @@ def _indico_email(value):
                 raise ValidationError(_('Invalid email address'))
 
         return _indico_email
+
+
+class AccompanyingPersonsField(RegistrationFormBillableField):
+    name = 'accompanying_persons'
+    mm_field_class = fields.List(fields.Nested({
+        'first_name': fields.String(required=True),
+        'last_name': fields.String(required=True),
```

Comment:
This will still allow empty strings (required just means it has to be present); add `validate=not_empty`